libraries/APM_Control/AP_AutoTune.cpp

```diff
@@ -250,7 +250,23 @@ void AP_AutoTune::update(AP_PIDInfo &pinfo, float scaler, float angle_err_deg)
 #if HAL_LOGGING_ENABLED
     if (now - last_log_ms >= 40) {
         // log at 25Hz
-        struct log_ATRP pkt = {
+        const struct PACKED {
```

Comment:
I would have expected this to be deleted from where ever it is now.

Reply:
Ah, I remember now.  We can't move the definition of this into the .cpp file.  log_ATRP is included into ArduPlane's log structure:

```
SCB-WAF: [1166/1206] Compiling ArduPlane/mode.cpp../../ArduPlane/Log.cpp:363:41: error: 'log_ATRP' is not a member of 'AP_AutoTune'
SCB-WAF: ../../ArduPlane/Log.cpp:363:41: error: 'log_ATRP' is not a member of 'AP_AutoTune'  363 |     { LOG_ATRP_MSG, sizeof(AP_AutoTune::log_ATRP),
SCB-WAF:   363 |     { LOG_ATRP_MSG, sizeof(AP_AutoTune::log_ATRP),      |                                         ^~~~~~~~
SCB-WAF:       |                                         ^~~~~~~~compilation terminated due to -Wfatal-errors.
```

... I discovered that but only half-undid the change :-(

I really think we should look at changing this to a WriteStreaming or similar...